packages/vite/src/node/optimizer/index.ts

```diff
@@ -242,7 +243,16 @@ export async function optimizeDeps(
     const flatId = flattenId(id)
     flatIdDeps[flatId] = deps[id]
     const entryContent = fs.readFileSync(deps[id], 'utf-8')
-    const exportsData = parse(entryContent) as ExportsData
+    let exportsData: ExportsData
+    try {
+      exportsData = parse(entryContent) as ExportsData
+    } catch {
+      config.logger.warn(
+        `Unable to parse dependency: ${id}. Trying again with an esbuild transform.`
+      )
+      const transformed = await transform(entryContent, config.esbuild || {})
```

Comment:
Why not use `{ loader: { '.js': 'jsx' } }` automatically, removing the need for user to configure it?

Reply:
That's a good question. Even if we hardcode this here, you still do need the following to make it work, because it would otherwise fail at a later stage when optimising deps:
```js
        optimizeDeps: {
          esbuildOptions: {
            loader: {
              ".js": "jsx",
            },
          },
        },
```

Reply:
We could move this statement up:
https://github.com/vitejs/vite/blob/5935055c5a9731910bd6f0f14c8c02cd5ecbb082/packages/vite/src/node/optimizer/index.ts#L266-L267
..then mutate the `esbuildOptions` from here:
```js
esbuildOptions.loader ??= {}
esbuildOptions.loader['.js'] ??= 'jsx'
const transformed = await transform(entryContent, esbuildOptions)
```

Reply:
Nice idea. I did that, although couldn't directly use `esbuildOptions` for the `transform` call (since the `loader` field type isn't compatible). Let me know what you think!